test/source/tests/unit-node.ts

```diff
@@ -1634,6 +1634,142 @@ jA==
       t.pass();
     });
 
+    ava.default('[unit][KeyUtil.decrypt] validates the private key', async t => {
```

Comment:
This looks good. Does this also fix the issue as experienced when private key is loaded:
 - from inbox backup
 - from EKM

Reply:
In other words, is this a full / verified fix for the problem?

Reply:
Let me add validation for `KeyUtil.readBinary()` too...

Reply:
Done